src/lib/validateRequest.ts

```diff
@@ -1,17 +1,67 @@
-import {InvalidRequestException} from './exceptions'
+import {UnauthorizedException} from './exceptions'
 import {decryptCookie} from './cookieEncrypter'
 
-export default function validateRequest(csrfHeader: string | undefined, csrfCookie: string | undefined, originHeader: string | undefined, allowedOrigins: string[], encKey: string) {
-    if (allowedOrigins.findIndex((value) => value === originHeader) == -1) {
-        throw new InvalidRequestException()
+export default function validateRequest(data: ValidateRequestData, options: ValidateRequestOptions) {
+
+    if (options.requireTrustedOrigin) {
+        if (data.allowedOrigins.findIndex((value) => value === data.originHeader) == -1) {
+            
+            const error = new UnauthorizedException()
+            error.logInfo = 'The call is from an untrusted web origin'
```

Comment:
I think we should log the origin header. Maybe also contents of `allowedOrigins`?